### PR TITLE
SPI delays for maxim platform

### PR DESCRIPTION
**Pull request by CiprianRegus**

### CODE REVIEW

File: drivers/platform/maxim/max32650/maxim_spi.c [hunk 1]
```diff
@@ -204,23 +283,31 @@ int32_t max_spi_init(struct no_os_spi_desc **desc,
 	int32_t ret;
 	struct no_os_spi_desc *descriptor;
 	struct max_spi_init_param *eparam;
+	struct max_spi_state *st;
 
 	if (!param || !param->extra)
 		return -EINVAL;
 
 	descriptor = calloc(1, sizeof(*descriptor));
-
```
Comment (amiclaus) [inline drivers/platform/maxim/max32650/maxim_spi.c:212]: nit: unrelated change

File: drivers/platform/maxim/max32650/maxim_spi.c [hunk 2]
```diff
@@ -297,12 +386,14 @@ int32_t max_spi_transfer(struct no_os_spi_desc *desc,
 		req.txLen = req.txData ? msgs[i].bytes_number : 0;
 		req.rxLen = req.rxData ? msgs[i].bytes_number : 0;
 
+		_max_delay_config(desc, &msgs[i]);
 		ret = MXC_SPI_MasterTransaction(&req);
-
```
Comment (amiclaus) [inline drivers/platform/maxim/max32650/maxim_spi.c:301]: ditto

File: drivers/platform/maxim/max32655/maxim_spi.c [hunk 3]
```diff
@@ -120,23 +201,31 @@ int32_t max_spi_init(struct no_os_spi_desc **desc,
 	int32_t ret;
 	struct no_os_spi_desc *descriptor;
 	struct max_spi_init_param *eparam;
+	struct max_spi_state *st;
 
 	if (!param || !param->extra)
 		return -EINVAL;
 
 	descriptor = calloc(1, sizeof(*descriptor));
-
```
Comment (amiclaus) [inline drivers/platform/maxim/max32655/maxim_spi.c:128]: ditto

File: drivers/platform/maxim/max32655/maxim_spi.c [hunk 4]
```diff
@@ -213,12 +304,14 @@ int32_t max_spi_transfer(struct no_os_spi_desc *desc,
 		req.txLen = req.txData ? msgs[i].bytes_number : 0;
 		req.rxLen = req.rxData ? msgs[i].bytes_number : 0;
 
+		_max_delay_config(desc, &msgs[i]);
 		ret = MXC_SPI_MasterTransaction(&req);
-
```
Comment (amiclaus) [inline drivers/platform/maxim/max32655/maxim_spi.c:217]: ditto

File: drivers/platform/maxim/max32660/maxim_spi.c [hunk 5]
```diff
@@ -110,23 +189,31 @@ int32_t max_spi_init(struct no_os_spi_desc **desc,
 	int32_t ret;
 	struct no_os_spi_desc *descriptor;
 	struct max_spi_init_param *eparam;
+	struct max_spi_state *st;
 
 	if (!param || !param->extra)
 		return -EINVAL;
 
 	descriptor = calloc(1, sizeof(*descriptor));
-
```
Comment (amiclaus) [inline drivers/platform/maxim/max32660/maxim_spi.c:118]: ditto

File: drivers/platform/maxim/max32660/maxim_spi.c [hunk 6]
```diff
@@ -203,12 +292,14 @@ int32_t max_spi_transfer(struct no_os_spi_desc *desc,
 		req.txLen = req.txData ? msgs[i].bytes_number : 0;
 		req.rxLen = req.rxData ? msgs[i].bytes_number : 0;
 
+		_max_delay_config(desc, &msgs[i]);
 		ret = MXC_SPI_MasterTransaction(&req);
-
```
Comment (amiclaus) [inline drivers/platform/maxim/max32660/maxim_spi.c:207]: ditto

File: drivers/platform/maxim/max32665/maxim_spi.c [hunk 7]
```diff
@@ -199,23 +278,31 @@ int32_t max_spi_init(struct no_os_spi_desc **desc,
 	int32_t ret;
 	struct no_os_spi_desc *descriptor;
 	struct max_spi_init_param *eparam;
+	struct max_spi_state *st;
 
 	if (!param || !param->extra)
 		return -EINVAL;
 
 	descriptor = calloc(1, sizeof(*descriptor));
-
```
Comment (amiclaus) [inline drivers/platform/maxim/max32665/maxim_spi.c:207]: ditto

File: drivers/platform/maxim/max78000/maxim_spi.c [hunk 8]
```diff
@@ -118,25 +199,33 @@ int32_t max_spi_init(struct no_os_spi_desc **desc,
...
 	int32_t ret;
+	struct max_spi_state *st;
 	struct no_os_spi_desc *descriptor;
 	struct max_spi_init_param *eparam;
 
 	if (!param || !param->extra)
 		return -EINVAL;
 
 	descriptor = calloc(1, sizeof(*descriptor));
-
```
Comment (amiclaus) [inline drivers/platform/maxim/max78000/maxim_spi.c:128]: ditto

File: drivers/platform/maxim/max78000/maxim_spi.c [hunk 9]
```diff
@@ -214,12 +305,14 @@ int32_t max_spi_transfer(struct no_os_spi_desc *desc,
 		req.txLen = req.txData ? msgs[i].bytes_number : 0;
 		req.rxLen = req.rxData ? msgs[i].bytes_number : 0;
 
+		_max_delay_config(desc, &msgs[i]);
 		ret = MXC_SPI_MasterTransaction(&req);
-
```
Comment (amiclaus) [inline drivers/platform/maxim/max78000/maxim_spi.c:218]: ditto

**Comment by CiprianRegus** [comment]
Created a separate commit for the extra lines removal.

**Comment by buha** [comment]
question here: how do we let developers know that on their platform, the cs_change_delay, cs_delay_first, cs_delay_last are ignored ? currently, they are not being ignored only on maxim platform

**Comment by CiprianRegus** [comment]
Added compile time warnings about SPI delays not being supported on some platforms.

**Review by buha — APPROVED** [review]
